includes/Core/Ecommerce/Requests/Orders.php

```diff
@@ -79,4 +79,30 @@ protected function clear_revenue_cookie( $response ) {
 
         return $response;
     }
+
+
+    /**
+     * @param $order_id
+     * @return bool
+     */
+    protected function is_first_order( $order_id ) {
+        $order = wc_get_order( $order_id );
+
+        $user_id = $order->get_user_id();
+
+        $args = [
+            'limit' => 1,
+            'return' => 'ids',
+        ];
+
+        if ( $user_id ) {
+            $args['customer_id'] = $user_id;
+        } else {
+            $args['billing_email'] = $order->get_billing_email();
+        }
+
+        $customer_order = wc_get_orders( $args );
+
```

Comment:
I think this will always return `true`. Cause you are trying to find customer orders using limit 1 without excluding this current order.